src/infrastructure/RequestHelper.ts

```diff
@@ -1,150 +1,97 @@
-import FormData from 'form-data';
-import { decamelizeKeys } from 'humps';
-import { stringify } from 'query-string';
-import { skipAllCaps } from './Utils';
+import Li from 'li';
 import {
   PaginatedRequestOptions,
   BaseRequestOptions,
-  DefaultRequestOptions,
   GetResponse,
   PostResponse,
   PutResponse,
   DelResponse,
 } from '../../types/types';
+import { BaseService } from './BaseService';
 
-function defaultRequest(service, endpoint: string, { body, query, sudo }: DefaultRequestOptions) {
-  return [
-    endpoint,
-    {
-      baseUrl: service.url,
-      headers: { sudo, ...service.headers },
-      query: query && stringify(decamelizeKeys(query), { arrayFormat: 'bracket' }),
-      body: body && decamelizeKeys(body, skipAllCaps),
-      rejectUnauthorized: service.rejectUnauthorized,
-      json: true,
-    },
-  ];
-}
-
-async function getImplementation(
-  service,
+export async function get(
+  service: BaseService,
   endpoint: string,
   options: PaginatedRequestOptions = {},
-  isNextPageRequest: boolean = false,
 ): Promise<GetResponse> {
   const { showPagination, maxPages, sudo, ...query } = options;
-  const requestOptions = defaultRequest(service, endpoint, {
+  const { headers, body } = await service.requester.get(service, endpoint, {
     query,
     sudo,
   });
 
-  const { headers, body } = await service.requester.get(...requestOptions);
-
   const pagination = {
     total: headers['x-total'],
-    next: headers['x-next-page'] || null,
-    current: headers['x-page'] || null,
+    next: parseInt(headers['x-next-page'], 10) || null,
+    current: parseInt(headers['x-page'], 10) || 1,
     previous: headers['x-prev-page'] || null,
     perPage: headers['x-per-page'],
     totalPages: headers['x-total-pages'],
   };
 
   const underLimit = maxPages ? pagination.current < maxPages : true;
 
-  // If not looking for a singular page and still under the max pages limit
-  // AND their is a next page, paginate
-  if ((isNextPageRequest || !query.page) && underLimit && pagination.next) {
-    const more = await getImplementation(
-      service,
-      endpoint,
-      {
-        ...options,
-        page: pagination.next
-      },
-      true,
-    );
+  if (!query.page && underLimit && pagination.next) {
+    const { next } = Li.parse(headers.link);
+    const more = await get(service, next.replace(/(.+\/api\/v[0-9]\/)/, ''), { maxPages, sudo });
```

Comment:
Reduces the complexity and avoid capturing group which is unnecessarily expensive especially when unused :laughing: 
```suggestion
    const more = await get(service, next.replace(/.+\/api\/v\d\//, ''), { maxPages, sudo });
```

Reply:
I like the change! ill commit it once i get better wifi